independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/AugmentAction.java

```diff
@@ -3,6 +3,9 @@
 import java.util.Set;
 
 public interface AugmentAction {
+
+    void performCustomBuild(String resultConsumer, Object context, String... finalOutputs);
```

Comment:
Can we have some comments on what this does as it's not really clear by the name?